watch-library/hardware/watch/watch_private.c

```diff
@@ -245,43 +245,35 @@ void _watch_enable_usb(void) {
 
 // this function ends up getting called by printf to log stuff to the USB console.
 int _write(int file, char *ptr, int len) {
+    int written = 0;
     (void)file;
-    if (hri_usbdevice_get_CTRLA_ENABLE_bit(USB)) {
-        tud_cdc_n_write(0, (void const*)ptr, len);
-        tud_cdc_n_write_flush(0);
-        return len;
+
+    while (hri_usbdevice_get_CTRLA_ENABLE_bit(USB) && written == 0) {
```

Comment:
Previously, if you tried to write something large it would write whatever it could (tud_cdc_n_write) and then claim it had written the whole buffer.

Unfortunately, https://github.com/bminor/newlib/blob/5192d5ea51dee786ba1cb97b216bd40115dbbf20/newlib/libc/stdio/fflush.c#L225 demands that we actually write something othewise it's an error, hence there's a busy loop here waiting for our buffer to clear. This is presumably horribly inefficient, but I'm not sure how to do it the right way.

Reply:
So, experimenting with do both USB CDC and mass storage at the same time, this blocking operation causes us to freeze up if we spam debug statements in the USB mass storage callbacks (probably because we're already busy with mass storage and can't write CDC things at the same time).

Not sure whether the lesson is 'don't do silly things in callbacks' or 'we need an intermediate buffer here' or ...